### ISSUE TITLE
Add support with youtube-dl

### ISSUE THREAD

**Issue by meav** [darsain/uosc#9]
I have few suggestion to enhance this script:

- Add title in playlist with playlist played by youtube-dl, like script `titleresolver.lua` in this reposity: [https://github.com/jonniek/mpv-playlistmanager](url)

With `playlistmanage.lua` and `titleresolver.lua`:

![1](https://user-images.githubusercontent.com/29623675/80196961-38fb9300-8648-11ea-9d11-10065c0e9caf.png)

With your `uosc.lua`:

![1](https://user-images.githubusercontent.com/29623675/80197094-621c2380-8648-11ea-8576-8972ca26b668.png)

- Add choose resolution with youtube-dl like this scirpt: [https://github.com/jgreco/mpv-youtube-quality](url)

Sorry for my bad english and thank you for reading

**Comment by darsain** [comment]
Yes, stream quality is already on my wish-list. I'll look into playlist tittles when working on it as well. Dunno when I'll have time for it though, so no ETA.

**Comment by FichteFoll** [comment]
Also note that caching the media title as a first measure and without calling `youtube-dl` would already be a pretty useful improvement. See https://github.com/jonniek/mpv-playlistmanager/issues/25 for an issue describing this feature.

**Comment by FichteFoll** [comment]
Additionally, mpv honors playlist title overrides from [extended m3u files](https://en.wikipedia.org/wiki/M3U#Extended_M3U) as shown in https://github.com/beetbox/beets/issues/3595#issuecomment-629869671. Neither playlistmanager nor uosc can do this currently.

**Comment by darsain** [comment]
For now I at least added a very simple stream quality switching menu, which just assigns `ytdl-format`, reloads the video, and resumes playback. No fetching of available formats going on.

What values show up there is configured in `stream_quality_options` option.

The format used is:

```
bestvideo[height<=?{height}]+bestaudio/best[height<=?{height}]
```

If it can be improved in some way, let me know.

Try it out on [dev branch](https://github.com/darsain/uosc/tree/dev).
